pkg/component/component.go

```diff
@@ -856,8 +856,33 @@ func GetComponentNames(client *occlient.Client, applicationName string) ([]strin
 	return names, nil
 }
 
-// List lists components in active application
-func List(client *occlient.Client, applicationName string, localConfigInfo *config.LocalConfigInfo) (ComponentList, error) {
+func ListDevfileComponents(client *occlient.Client, selector string) (ComponentList, error) {
```

Comment:
+1 especially description for selector. 
It looks a little bit weird to me as we don't use selectors for odo components.

Reply:
yes @kadel  - we use the project selector when we want to have components from all application, but in case of s2i components - we list all applications and then list per component application 
https://github.com/openshift/odo/blob/e5798c42c9ce191425e3ed18e52db985e1e2f920/pkg%2Fodo%2Fcli%2Fcomponent%2Flist.go#L254-L261

Reply:
Now the S2I approach is not optimised as it Xn calls to the where n is the number of applications and X is the calls done per application.